packages/framework/aqueduct/src/data-object-factories/pureDataObjectFactory.ts

```diff
@@ -58,10 +59,9 @@ export interface IFluidDataObjectFactory {
      * They get attached only when a handle to one of them is attached to already attached objects.
      */
     createChildInstance<
-        O,
-        S,
-        TObject extends PureDataObject<O,S>,
-        TFactory extends PureDataObjectFactory<TObject, O,S>>
+        TObject extends PureDataObject<O, S, E>,
+        TFactory extends PureDataObjectFactory<TObject, O, S, E>,
+        O, S, E extends IEvent = IEvent>
```

Comment:
curious, is typescript able to infer these types since they already exist on TObject and TFactory?

Reply:
Yep, it can.
This same technique does not work for interfaces though, as there there is no "construction" or call with concreate type to extract information from